src/mongoc/mongoc-write-concern.c

```diff
@@ -205,6 +205,7 @@ mongoc_write_concern_set_wtimeout (mongoc_write_concern_t *write_concern,
                                    int32_t            wtimeout_msec)
 {
    bson_return_if_fail(write_concern);
+   bson_return_if_fail(wtimeout_msec >= 0);
```

Comment:
That's a great question. I think you should not use bson_return_if_fail, even though using it is consistent with the surrounding code. In this case, consistency with bad code is bad. As long as we are improving the driver by validating wtimeout_msec, and improving it further by adding good tests, let's be inconsistent: if wtimeout_msec < 0, MONGOC_WARN the user and refuse to set the value.

If you would like to add a test framework feature to catch and verify the warning instead of printing it to stderr, that'd be awesome. Otherwise I can come in after you and do that.

Eventually CDRIVER-697 will require a wholesale replacement of bson_return_if_fail to validate user input, along with tests of the improved validation. In the meantime, let's head in that direction incrementally instead of adding more bad uses of bson_return_if_fail.